### PR TITLE
feat(tests): add session-scoped cleanup with optional --no-clean mode

### PR DESCRIPTION
**Pull request by yael-avioz-thenvoi**
## Summary
- Add `--no-clean` pytest option and `THENVOI_TEST_NO_CLEAN` env var to skip test cleanup
- Add `is_no_clean_mode()` helper function for fixtures to check cleanup mode
- Update `test_dynamic_agent` and `test_participant_permissions` fixtures with inline cleanup logic

## Use Cases
No-clean mode is useful for:
- Testing user limits (max agents, max chats)
- Finding edge case bugs with accumulated data
- Debugging test failures by inspecting created resources

## Usage
```bash
# Skip cleanup via flag
uv run pytest tests/integration/ -v --no-clean

# Skip cleanup via env var
THENVOI_TEST_NO_CLEAN=1 uv run pytest tests/integration/ -v
```

## Implementation
Simple per-fixture cleanup pattern (~10 lines per fixture):
```python
@pytest.fixture(scope="module")
async def my_fixture(client, request):
    resource = await create_resource()
    yield resource
    
    if not is_no_clean_mode(request):
        await delete_resource(resource)
    else:
        logger.info("[NO-CLEAN MODE] Skipping cleanup")
```

## Test plan
- [x] All unit tests pass
- [x] Ruff lint and format checks pass
- [x] Unit tests for `is_no_clean_mode()` helper (10 tests)
- [ ] Integration tests with cleanup enabled (default behavior)
- [ ] Integration tests with `--no-clean` flag

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by amit-gazal-thenvoi — COMMENTED** [review]
## Code Review

This PR is well-documented and thoroughly tested. However, I have some feedback and a suggestion for a simpler approach.

---

## 🤔 Architectural Suggestion: Consider a Simpler Approach

If the main goal is `--no-clean` support for debugging, this could be achieved with ~10 lines per fixture instead of ~750 new lines:

```python
@pytest.fixture(scope="module")
async def dynamic_agent(module_user_api_client, request):
    # ... create agent ...
    yield _dynamic_agent
    
    # Simple flag check in existing cleanup
    if not is_no_clean_mode(request):
        await module_user_api_client.human_api.delete_my_agent(id=agent.id, force=True)
```

### Comparison

| Feature | Simple Flag Check | Centralized Tracker |
|---------|-------------------|---------------------|
| `--no-clean` support | ✅ | ✅ |
| Cleanup timing | Per-module | Session-end |
| Code complexity | ~5 lines/fixture | ~750 new lines |
| Thread-safety concerns | None | Yes (singleton) |

### Questions to Consider
1. Do tests across modules need to share dynamically created agents?
2. Is session-scoped cleanup (vs module-scoped) actually needed?
3. Is the resource summary at session end worth the added complexity?

If the answers are "no", the simpler approach would achieve the goal with significantly less code.

File: tests/conftest_integration.py [hunk 1]
```diff
@@ -198,3 +421,126 @@ async def test_peer_id(api_client: AsyncRestClient | None) -> str | None:
...
+    """
+    yield  # Run all tests first
+
+    # Check if cleanup should be skipped
+    no_clean = is_no_clean_mode(request)
+    tracker = resource_tracker
+
+    # Print summary (using print() instead of logging for visibility without -s flag)
+    if tracker.agents:
+        print("=" * 60)
```
Comment (amit-gazal-thenvoi) [inline tests/conftest_integration.py]: **`print()` statements need `-s` flag anyway**

The comment above explains the rationale (visibility without `-s` flag), but pytest actually captures stdout by default, so users still need `-s` to see this output.

Consider:
- Using `pytest.config.hook.pytest_report_header` or similar pytest hooks for more reliable output
- Or document that `-s` is recommended when running integration tests

**Review by amit-gazal-thenvoi — APPROVED** [review]
Looks great
